Mathlib/Data/Real/EReal.lean

```diff
@@ -929,6 +929,94 @@ lemma neg_sub {x y : EReal} (h1 : x ≠ ⊥ ∨ y ≠ ⊥) (h2 : x ≠ ⊤ ∨ y
     - (x - y) = - x + y := by
   rw [sub_eq_add_neg, neg_add _ _, sub_eq_add_neg, neg_neg] <;> simp_all
 
+/-! ### Addition and order -/
+
+lemma le_iff_le_forall_real_gt (x y : EReal) : (∀ z : ℝ, x < z → y ≤ z) ↔ y ≤ x := by
```

Comment:
This and the next lemma should follow much more easily from properties about `WithTop` and `WithBot`. Can you try to reprove them in a simpler way using those instead? If you can't manage it, ping me and I'll give it a go.

Reply:
@j-loreaux : thanks for the idea. Here is what I've done:

- new lemmas `le_forall_gt_iff_le_of_withBot`, `ge_forall_lt_iff_ge_of_withBot`,  `le_forall_gt_iff_le_of_withTop`, `ge_forall_lt_iff_ge_of_withTop` in Order.WithBot. The idea is that the lemmas `le_of_forall_le_of_dense`, `ge_of_forall_le_of_dense` need the space to have no Bot/Top element, but we can actually push them through WithBot/WithTop constructions!
- the versions for EReal `le_forall_real_gt_iff_le`, `ge_forall_real_lt_iff_ge` can be deduced from `le_forall_gt_iff_le_of_withBot` pretty easily (I actually don't need the three other versions of the latter). This is not completely trivial (EReal is _WithBot (WithTop R)_, so `le_forall_gt_iff_le_of_withBot` gets you a version with _WithTop R_ instead of _R_), but still much cleaner than the current state of things.

Does this looks good to you?

There are two specific points on which I'd like your feedback:

- I am still quite bad at naming lemmas. I don't really like this version (for instance `le_forall_gt_iff_le_of_withBot`), so any suggestion is welcome.
- the proof of `ge_forall_real_lt_iff_ge` is too cumbersome. It should be a one-liner using `le_forall_real_gt_iff_le` and the negation (as an order-reversing isomorphism), but I don't know how to write it.